src/connectors/hits/connectHits.ts

```diff
@@ -113,7 +113,9 @@ const connectHits: HitsConnector = function connectHits(
           false
         );
 
-        renderState.sendEvent('view', renderState.hits);
+        if (!renderOptions.searchMetadata.isSearchStalled) {
+          renderState.sendEvent('view', renderState.hits);
+        }
```

Comment:
Could we rather move all the conditionals to [`createSendEventForHits()`](https://github.com/algolia/instantsearch.js/blob/0a517609de103eef4f8edfefe6e28a1d79a14209/src/lib/utils/createSendEventForHits.ts#L150)?

Reply:
Sure, that's possible

Reply:
Would it make sense here? I'm not convinced that a function responsible for transformation should also take global state in account: 

https://github.com/algolia/instantsearch.js/blob/99f6fe1dc51e4815e5b9efcfb30e3e2f3127e763/src/lib/utils/createSendEventForHits.ts#L91-L104

Reply:
I thought about having a short circuit before this line: https://github.com/algolia/instantsearch.js/blob/99f6fe1dc51e4815e5b9efcfb30e3e2f3127e763/src/lib/utils/createSendEventForHits.ts#L159

But I don't have the complete knowledge of this part.

> Also note that the sendEvent function can be called by end users. I'm not sure if we then also should prevent forwarding the event if the search is currently stalled

I think the question is: should it send an event when the search is stalled?

I don't think it should because (IIUC):

1. a render displays hits: send view event
2. user searches
3. a render displays hits again because search is stalled: ~send view event~

And so I don't think this (3) makes sense.

Reply:
if we do this condition only for `view` event I agree, but if it should be all events that get shortcutted I don't think so, because you could be in a stalled state, but click on a buy button, that's a genuine action which should count right?

Reply:
I think it makes sense to have guards per event type centralized in `createSendEventForHits`. For now we can have only one for `view` since we have knowledge about when it should and should not be sent, we don't need to guard for all events.